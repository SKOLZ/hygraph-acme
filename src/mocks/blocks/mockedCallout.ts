export const mockedCallout = {
  __typename: "Callout" as const,
  id: "clt7bqmp49dkq07uj23atb9h5",
  title: "Callout title",
  description: {
    raw: {
      children: [
        {
          type: "paragraph",
          children: [
            {
              text: "Callout description",
            },
          ],
        },
      ],
    },
  },
  actionButton: {
    buttonText: "Callout button",
    size: null,
  },
};
