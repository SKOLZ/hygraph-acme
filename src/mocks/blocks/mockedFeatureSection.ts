import { BlockFeatureSection, Direction, Sizes } from "@/generated/graphql";

export const mockedFeatureSection = {
  __typename: "FeatureSection" as const,
  id: "clt7bqmp09dkf07uj2k6in82a",
  label: "feature section label",
  title: "feature section title",
  description: {
    raw: {
      children: [
        {
          type: "paragraph",
          children: [
            {
              text: "As a composable and headless platform, Acme works with all your preferred frontend frameworks, existing systems, GraphQL and REST APIs.",
            },
          ],
        },
      ],
    },
  },
  media: {
    __typename: "StaticImage" as const,
    image: {
      url: "https://media.graphassets.com/bbd3PKNcTBGVkNbcois7",
      width: 480,
      height: 320,
    },
    altText: "Stats page showing a profile views chart and a chat window",
  },
  direction: Direction.LeftToRight,
  featureSectionButton: {
    buttonText: "Get started",
    size: Sizes.Lg,
  },
};
