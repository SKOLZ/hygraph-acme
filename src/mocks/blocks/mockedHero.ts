import { BlockHero, Sizes } from "@/generated/graphql";

export const mockedHero = {
  __typename: "Hero" as const,
  id: "1",
  title: `Hero title`,
  description: {
    raw: {
      children: [
        {
          type: "paragraph",
          children: [
            {
              text: "Hero description",
            },
          ],
        },
      ],
    },
  },
  buttonActions: [
    {
      id: "clt7bqmoz9dkc07ujrs01ghn3",
      buttonText: "Get started",
      size: Sizes.Lg,
    },
  ],
};
