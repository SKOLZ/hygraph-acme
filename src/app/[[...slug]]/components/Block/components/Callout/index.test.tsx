import { render } from "@testing-library/react"
import { Callout } from ".";
import { mockedCallout } from "@/mocks/blocks/mockedCallout";

describe("Callout", () => {
  it("renders the component correctly", () => {
    const { container } = render(<Callout {...mockedCallout} />);
    expect(container).toMatchSnapshot();
  })
})
