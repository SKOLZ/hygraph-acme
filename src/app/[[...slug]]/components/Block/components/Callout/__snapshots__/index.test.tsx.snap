// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Callout renders the component correctly 1`] = `
<div>
  <div
    class="mx-auto w-full max-w-screen-xl px-8"
  >
    <div
      class="flex flex-col items-start justify-between gap-8 rounded-2xl bg-secondary-800 p-8 xl:flex-row xl:p-16"
    >
      <div
        class="flex flex-col items-start gap-4"
      >
        <h2
          class="title-2 text-white"
        >
          Callout title
        </h2>
        <p
          class="text-1 text-white"
        >
          Callout description
        </p>
      </div>
      <a
        class="button button-md "
        href="#"
      >
        Callout button
      </a>
    </div>
  </div>
</div>
`;
